06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * InvertedIndex class
+ * @class
+ */
+class InvertedIndex {
+  /**
+   * class constructor
+   * @constructor
+   */
+  constructor() {
+    this.files = [];
+    this.showAllFiles = false;
+    this.indexed = {};
+    this.showAllFilesSearch = false;
+    this.searchAllResult = {};
+  }
+
+  /**
+   * Validates file content
+   * @function
+   * @param {Array} fileContent content of uploaded file
+   * @return {Array} valid fileContent or false if invalid file
+   */
+  static validateFile(fileContent) {
+    if (fileContent.constructor !== Array || fileContent.length < 1) {
+      return false;
+    }
+    let invalid = false;
+    fileContent.forEach((book) => {
+      if (book.title === undefined || book.text === undefined) {
+        invalid = true;
+        return;
+      }
+      if (book.text.length < 1) invalid = true;
+    });
+    if (invalid) return false;
```

Comment:
You JSDoc says it expects an `Array` but you're returning a boolean here. Let's capture this also please.

Note: Fix this in all places it applies in your code.